src/internal/connector/onedrive/item.go

```diff
@@ -61,18 +62,55 @@ func oneDriveItemMetaReader(
 	service graph.Servicer,
 	driveID string,
 	item models.DriveItemable,
-) (io.ReadCloser, int, error) {
-	meta, err := oneDriveItemMetaInfo(ctx, service, driveID, item)
-	if err != nil {
-		return nil, 0, err
+	fetchPermissions bool,
+) (r io.ReadCloser, size int, err error) {
```

Comment:
Just wondering if we should have a conversation about using naked returns in the codebase.

From a quick grep, it looks like we only use naked returns when we deal with `recover()` as of now:
```
$ rg '[^A-z]\([^\(]*,?\w+ \w+\) \{'
# recover
internal/operations/backup.go:func (op *BackupOperation) Run(ctx context.Context) (err error) {
internal/operations/restore.go:func (op *RestoreOperation) Run(ctx context.Context) (restoreDetails *details.Details, err error) {

# tests (not naked returns though)
internal/connector/uploadsession/uploadsession_test.go:func (mr *mockReader) Read(b []byte) (n int, err error) {
internal/kopia/model_store_test.go:             func(innerCtx context.Context, w repo.RepositoryWriter) (innerErr error) {

# not naked returns even though named
internal/kopia/upload.go:func (rw *backupStreamReader) Read(p []byte) (n int, err error) {
internal/kopia/upload.go:func (rw *restoreStreamReader) Read(p []byte) (n int, err error) {
```

Reply:
My vote goes to always using naked returns, unless required by a recover.  Post processing in defer can be similarly achieved by separating code into two funcs, with less confusion to boot.